mm2src/mm2_main/tests/docker_tests/solana_tests.rs

```diff
@@ -131,14 +143,314 @@ fn test_disable_solana_platform_coin_with_tokens() {
+    assert_eq!(
+        "FJktmyjV9aBHEShT4hfnLpr9ELywdwVtEL1w1rSWgbVf",
+        solana_key_pair.pubkey().to_string()
+    );
+
+    let other_solana_keypair = generate_key_pair_from_iguana_seed("bob passphrase".to_string());
+    assert_eq!(
+        "B7KMMHyc3eYguUMneXRznY1NWh91HoVA2muVJetstYKE",
+        other_solana_keypair.pubkey().to_string()
+    );
+}
+
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn solana_transaction_simulations_dockerized() {
+    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+    let (_, sol_coin) = solana_coin_for_test(passphrase, SolanaNet::Local);
+    let request_amount = BigDecimal::try_from(0.0001).unwrap();
+    let valid_tx_details = block_on(
+        sol_coin
+            .withdraw(WithdrawRequest::new(
+                "SOL".to_string(),
+                None,
+                sol_coin.my_address.clone(),
+                request_amount.clone(),
+                false,
+                None,
+                None,
+            ))
+            .compat(),
+    )
+    .unwrap();
+    let (_, fees) = block_on(sol_coin.estimate_withdraw_fees()).unwrap();
+    let sol_required = lamports_to_sol(fees);
+    let expected_spent_by_me = &request_amount + &sol_required;
+    assert_eq!(valid_tx_details.spent_by_me, expected_spent_by_me);
+    assert_eq!(valid_tx_details.received_by_me, request_amount);
+    assert_eq!(valid_tx_details.total_amount, expected_spent_by_me);
+    assert_eq!(valid_tx_details.my_balance_change, sol_required.neg());
+}
+
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn solana_transaction_zero_balance_dockerized() {
+    let passphrase = "fake passphrase".to_string();
+    let (_, sol_coin) = solana_coin_for_test(passphrase, SolanaNet::Local);
+    let invalid_tx_details = block_on(
+        sol_coin
+            .withdraw(WithdrawRequest::new(
+                "SOL".to_string(),
+                None,
+                sol_coin.my_address.clone(),
+                BigDecimal::from_str("0.000001").unwrap(),
+                false,
+                None,
+                None,
+            ))
+            .compat(),
+    );
+    let error = invalid_tx_details.unwrap_err();
+    let (_, fees) = block_on(sol_coin.estimate_withdraw_fees()).unwrap();
+    let sol_required = lamports_to_sol(fees);
+    match error.into_inner() {
+        WithdrawError::NotSufficientBalance { required, .. } => {
+            assert_eq!(required, sol_required);
+        },
+        e => panic!("Unexpected err {:?}", e),
+    };
+}
+
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+fn solana_transaction_simulations_not_enough_for_fees_dockerized() {
+    let passphrase = "non existent passphrase".to_string();
```

Comment:
@laruh I've refactored the code to move all the repeated strings into constants. However, I'm concerned that this might have gone a bit too far (especially in test case scenarios). I feel it may have impacted readability in some areas. Let me know if you think we should adjust this or leave it as it is right now.

Reply:
> I'm concerned that this might have gone a bit too far

Yeah, some consts are used just once
![Screenshot 2024-08-10 at 09 41 08](https://github.com/user-attachments/assets/531f4926-0e31-4c2f-b82f-260e0686f8be)

It was basically enough to add consts for strings mentioned in review comments
![Screenshot 2024-08-10 at 09 42 14](https://github.com/user-attachments/assets/40b1066a-924e-4e53-b77a-3f7becfc6322)

In spl_tests.rs you leave one `const USDC: &str = "USDC";`

In solana_common_tests consts are fine, just it would be better to rename `PASSPHRASE` to `SOL_PASSPHRASE` to note that its for solana, same for `ADDITIONAL_PASSPHRASE` -> `SOL_ADDITIONAL_PASSPHRASE`
https://github.com/KomodoPlatform/komodo-defi-framework/blob/00ecbb65b8924ebf56d6d0541b5829b5304e36f1/mm2src/mm2_main/tests/docker_tests/solana_common_tests.rs#L19-L20

@mj-blockydevs  please when you answer in the review that something was fixed, provide commit link/links to this fix.
Like in this example https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169#discussion_r1709478691 and below.
It is a lot easier to review fixes especially when we have notes from different reviewers and various fix commits.